aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -236,6 +236,56 @@ TORCH_PRECOMPUTE_META_FUNC(index_add)
   return TORCH_PRECOMPUTE_STRUCT(index_add)().set_dim(dim);
 }
 
+template <typename Meta>
+int64_t index_fill_meta_impl(
+    Meta& meta,
+    const Tensor& self,
+    int64_t dim,
+    const Tensor& index,
+    const Scalar& value) {
+
+  at::native::scatter_gather_dtype_check("index_fill", self, index);
+  at::assert_no_overlap(self, index);
+  if (at::has_internal_overlap(self) == at::MemOverlap::YES) {
+    TORCH_WARN(
+      "Use of index_fill on expanded tensors is deprecated. "
+      "Please clone() the tensor before performing this operation. "
+      "This also applies to advanced indexing e.g. tensor[mask] = scalar");
+  }
+  if (!self.is_complex() && value.isComplex()) {
+    TORCH_CHECK(false, "index_fill(): Converting complex Scalar to non-complex type is not supported");
+  }
+  TORCH_CHECK(index.dim() <= 1, "Index has to be a vector/scalar");
+
+  // auto& result = meta.maybe_get_output(0);
+  // Should we check assert_no_internal_overlap on result
+  // as it is done for index_add ?
+  meta.set_output(self.sizes(), self.options());
+
+  // TODO: add TI checks to meta as in index_add
```

Comment:
This TODO needs to be fixed. The construction of the TI needs to be done in the META function to get the TI checks here.